packages/x-data-grid/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -719,8 +720,10 @@ export const useGridColumnResize = (
       const columns = options.columns.map((c) => apiRef.current.state.columns.lookup[c]);
 
       try {
-        apiRef.current.unstable_setColumnVirtualization(false);
-        await columnVirtualizationDisabled();
+        if (!props.disableVirtualization && options.resizeAll) {
+          apiRef.current.unstable_setColumnVirtualization(false);
+          await columnVirtualizationDisabled();
+        }
```

Comment:
I'm not sure I follow, I thought this PR intended to disable *row* virtualization during auto resize 🤔
As you can see in the code, column virtualization is already disabled.

Reply:
so should I remove the column virtualization disabling logic along with this new option, because in default behavior it looks like we don't really need to disable column virtualization or we?

Reply:
I'm sorry, it's my bad. I thought this issue was about row virtualization not being disabled during autosizing, and [my comment above](https://github.com/mui/mui-x/pull/15116#pullrequestreview-2502334179) was referring to row virtualization as well.

Regarding column virtualization, as you pointed out, it works for column headers but doesn't work for actual cells. Your approach makes sense to me. Sorry about the confusion!

Reply:
However, we might need to add the `disableColumnVirtualization` option (defaulting to `true`).
The reason for this is that it's not always necessary.
For example, autosizing a single column on double-click – the horizontal scroll gets reset even though we don't need to disable column virtualization in this case: https://stackblitz.com/edit/react-4oxpjcvz?file=index.tsx

Reply:
thanks for the clarification, I have added `disableColumnVirtualization`